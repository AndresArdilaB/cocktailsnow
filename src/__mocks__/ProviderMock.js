import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { testState } from '../frontend/initialState';
import reducer from '../frontend/reducers';

const store = createStore(reducer, testState);
const history = createBrowserHistory();

const ProviderMock = ({ children }) => (
  <Provider store={store}>
    <Router history={history}>
      {children}
    </Router>
  </Provider>
);

export default ProviderMock;
