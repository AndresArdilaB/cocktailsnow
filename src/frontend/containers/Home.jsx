import React from 'react';
import { connect } from 'react-redux';
import Filters from '../components/Filters';
import CocktailDetail from '../components/CocktailDetail';
import '../assets/styles/App.styl';

const Home = ({ filters, currentCocktail }) => {

  return (
    <>
      <Filters filters={filters} isHome />
      <CocktailDetail data={currentCocktail} />
    </>
  );
};

const mapStateToProps = ({ filters, currentCocktail }) => {
  return {
    filters,
    currentCocktail,
  };
};

export default connect(mapStateToProps, null)(Home);
