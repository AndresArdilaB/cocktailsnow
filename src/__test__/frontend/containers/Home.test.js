import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../../__mocks__/ProviderMock';
import Home from '../../../frontend/containers/Home';

describe('<Home />', () => {

  test('Component render', () => {
    const home = shallow(
      <ProviderMock>
        <Home />
      </ProviderMock>,
    );

    expect(home.length).toEqual(1);
  });

});

describe('Home Snapshot', () => {

  test('Testing the home UI component', () => {
    const home = create(
      <ProviderMock>
        <Home />
      </ProviderMock>,
    );
    expect(home.toJSON()).toMatchSnapshot();
  });
});
