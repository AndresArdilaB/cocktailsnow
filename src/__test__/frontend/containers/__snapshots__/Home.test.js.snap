// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Snapshot Testing the home UI component 1`] = `
Array [
  <section
    className="filters__main"
  >
    <div
      className="filters__title"
    >
      Find your cocktail by some of this parameters
    </div>
    <section
      className="filters"
    >
      <div
        className="container"
      >
        <select
          className="select"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Categories
          </option>
          <option
            id="test"
            value="test"
          >
            Test
          </option>
        </select>
      </div>
      <div
        className="container"
      >
        <select
          className="select"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Glasses
          </option>
          <option
            id="test"
            value="test"
          >
            Test
          </option>
        </select>
      </div>
      <div
        className="container"
      >
        <select
          className="select"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Ingredients
          </option>
          <option
            id="test"
            value="test"
          >
            Test
          </option>
        </select>
      </div>
      <div
        className="container"
      >
        <select
          className="select"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Type
          </option>
          <option
            id="test"
            value="test"
          >
            Test
          </option>
        </select>
      </div>
    </section>
  </section>,
  <section
    className="detail"
  >
    <div
      className="title"
    >
      Random Cocktel
    </div>
    <div
      className="detail-container"
    >
      <div
        className="media"
      >
        <img
          alt="Cocktail"
          src="https://www.thecocktaildb.com/images/media/drink/qywpvt1454512546.jpg"
        />
      </div>
      <div
        className="text"
      >
        <div
          className="propierty"
          id="Lunch Box"
        >
          <h3>
            Name:
          </h3>
          <span>
            Lunch Box
          </span>
        </div>
        <div
          className="propierty"
        >
          <h3>
            Glass:
          </h3>
          <span>
            Pint glass
          </span>
        </div>
        <div
          className="propierty"
        >
          <h3>
            Category:
          </h3>
          <p>
            Beer
          </p>
        </div>
        <div>
          <h3>
            Ingredients:
          </h3>
          <span>
            Beer
          </span>
          <span>
            , Amaretto
          </span>
          <span>
            , Orange juice
          </span>
        </div>
        <div>
          <h3>
            Instructions:
          </h3>
          <p>
            Fill a pint glass almost full with beer. Then fill the rest with orange juice (careful not to fill it to the top). Then take the shot of Amaretto and drop it in.
          </p>
        </div>
      </div>
    </div>
  </section>,
]
`;
